### PR TITLE
Populate dataset

### PR DESCRIPTION
**Pull request by galvana**
# Purpose
Adds a `update_dataset` function which takes a `ConnectionConfig`, `DatasetConfig`, and API data to update the existing Dataset. This is meant to be used for SaaS connector development.

# Functionality
- Updates an existing dataset based on the API data
  - Can also write the generated dataset to a separate file from the existing dataset for comparisons
- The collection order is derived from the collection order in the SaaS config
  - Default behavior preserves the collection order of the existing dataset
- Preserves any user-defined data categories during the dataset generation
- Invoked from the `test_*_access_request_task` after the `run_access_request` has returned a response
  ```
  v = graph_task.run_access_request(
      privacy_request,
      policy,
      graph,
      [connection_config],
      {"email": identity_email},
  )

  update_dataset(
      connection_config,
      dataset_config,
      v,
      "generated_dataset.yml",
  )
  ```

# Checklist
- [x] Good unit test/integration test coverage
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

### CODE REVIEW

File: tests/test_helpers/dataset_utils.py
Comment (galvana) [inline tests/test_helpers/dataset_utils.py:137]: We have a `merge_datasets` function but we need to preserve the order of the fields and we lose control of that as soon as we convert a simple dict to a `FidesopsDataset`, the fields are alphabetized due to default Pydantic functionality.

File: tests/test_helpers/test_dataset_utils.py
Comment (adamsachs) [inline tests/test_helpers/test_dataset_utils.py:577]: would a better test here include an `api_data` object that has at least one of the existing collection elements placed in a different order than in `existing_dataset`?

File: tests/test_helpers/test_dataset_utils.py
Comment (adamsachs) [inline tests/test_helpers/test_dataset_utils.py:3]: nice job with these tests - they're a great starting point for understanding the expected functionality of the utilities.

one follow up - would it be feasible/useful to include more of an "integration" level test that actually tests and demonstrates how the utility can work on a _real_ API response, i.e. using a saas provider we already integrate with as an example? that test would obviously need to be marked as an `integration_saas` test since it would require external communication, but just seems like it could be a nice-to-have. (if you think this is feasible, maybe we can split out a separate issue/PR for it, since i don't think it should necessarily block this PR)

File: tests/test_helpers/dataset_utils.py
Comment (adamsachs) [inline tests/test_helpers/dataset_utils.py:76]: is this logic/processing covered in the unit tests? i know we're probably not going to be able to cover every piece of this, but it's some behavior that i think would be good to have covered both to ensure correct functionality and to clarify expected behavior

Comment (adamsachs) [reply tests/test_helpers/dataset_utils.py:76]: also, i think my suggestion about an "integration" level test may help in providing coverage for this piece of things? that would address my concern, no need to make it specifically a unit test.

maybe just another reason to look into having that sort of "integration" test with more realistic data

File: tests/test_helpers/dataset_utils.py
Comment (adamsachs) [inline tests/test_helpers/dataset_utils.py:107]: just noting that i don't totally follow this, but probably due to my lack of understanding, so may be good to review together offline

File: tests/test_helpers/dataset_utils.py
Comment (adamsachs) [inline tests/test_helpers/dataset_utils.py:227]: to me, it feels like this function should live more centrally. what would you think of defining it somewhere within `fidesops.graph.data_type.py`? from what i can tell, there's nothing yet defined there that takes an arbitrary input and gives the corresponding proper `Fidesops` `DataType`, but I think that's what you want here.

If you add that utility method over there, then I think here you should be relying upon the proper `DataType` enum - passing around string literals here just feels a bit sloppy to me. Specifically, I think it's ripe for unnoticed regressions if/when any of the typing logic changes in the fidesops "taxonomy" (for lack of a better term), and it will be burdensome to maintain.

File: tests/test_helpers/dataset_utils.py
Comment (adamsachs) [inline tests/test_helpers/dataset_utils.py:123]: not necessarily a feature for now, but a potential enhancement could be to allow for a configurable "sampling size" here so that potentially more than just the first row of response data is being used here. obviously we'd then need to build in some more advanced type-inference logic to take into account multiple values for each column, but just calling it out as an area for fine tuning.

**Comment by adamsachs** [comment]
it looks like there are some unit tests failing that i _think_ are real failures? once that's addressed, i think i'm good to approve - there is one additional nitpick i threw on there for the `get_data_type`, let me know what you think. i don't think it should block the PR but if you agree with it and it's a quick change, may as well get it in there.

**Review by adamsachs — APPROVED** [review]
nice, this all looks great - thanks for making those updates to move the data type logic more centrally, i think it will help for maintainability. 

excited to see this in action!

**Pull request merged**
